### PR TITLE
refactor(database): move user logic into separate package

### PR DESCRIPTION
**Pull request by jbrockopp**
Based off of https://github.com/go-vela/server/pull/574

This change continues the refactor effort initially introduced in the above PR.

This adds a new `user` package to the `github.com/go-vela/server/database` package.

This contains a `UserService` interface declaring all functions necessary for user based interactions with the database:

https://github.com/go-vela/server/blob/b2350a95513fedd31ac4f1915b5b2d91821226d1/database/user/service.go#L11-L45

This package also contains the `engine` which implements the above service interface:

https://github.com/go-vela/server/blob/b2350a95513fedd31ac4f1915b5b2d91821226d1/database/user/user.go#L25-L39

This `engine` contains no raw SQL queries for integrating with the `users` table.

Instead, we leverage our DB library's (https://gorm.io/) agnostic abstraction for integrating with that table.

### CODE REVIEW

**Review by cognifloyd — COMMENTED** [review]
Maybe point out in the docstrings ListUsers and ListLiteUsers differ (what makes it "Lite").
It looks like Lite just slips decrypting fields, right? Anything else?

Other than that, it looks good to me.

File: database/user/list_lite.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+package user
+
+import (
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/database"
+	"github.com/go-vela/types/library"
+)
+
+// ListLiteUsers gets a lite list of users from the database.
```
Comment (cognifloyd) [inline database/user/list_lite.go:13]:
```suggestion
// ListLiteUsers gets a lite (only: id, name) list of users from the database.
```

**Review by cognifloyd — COMMENTED** [review]
Maybe something like this?

**Review by cognifloyd — APPROVED** [review]

**Review by ecrupper — APPROVED** [review]
LGTM - really like this new pattern

**Pull request merged**